components/rangeslider/rangesliders/twoHandlesSliders/twoHandlesRangeSliderBase.js

```diff
@@ -0,0 +1,235 @@
+import { Components } from 'coherent-gameface-components';
+const components = new Components();
+import verticalTemplateTwoHandles from '../../templates/verticalTwoHandles.html';
+import horizontalTemplateTwoHandles from '../../templates/horizontalTwoHandles.html';
+// eslint-disable-next-line no-unused-vars
+import Rangeslider from '../../script';
+import RangeSliderBase from '../rangesliderBase';
+
+/**
+ * This class holds common methods and data for all the rangesliders
+ */
+export default class TwoHandlesRangeSliderBase extends RangeSliderBase {
+    /**
+     * @param {Rangeslider} rangeslider - The gameface-rangeslider custom element
+     */
+    constructor(rangeslider) {
+        super(rangeslider);
+        this.onMouseDown = this.onMouseDown.bind(this);
+        this.onMouseMove = this.onMouseMove.bind(this);
+
+        this.state = {
+            min: 0,
+            max: 100,
+            grid: false,
+            thumb: false,
+            step: 1,
+            value: [],
+            customHandleLeft: null,
+            customHandleRight: null,
+        };
+    }
+
+    // The derived class should override the next methods because they behave differently for basic and values range slider
+    // OVERRIDE START
+    /**
+     * Called when the element was attached to the DOM.
+     */
+    connectedCallback() { }
+    /**
+     * Sets the min and max value of the rangeslider
+     */
+    setMinAndMax() { }
+    /**
+     * Sets the value of the handles
+     */
+    setHandleValues() { }
+    /**
+     * Builds the grid
+     */
+    buildGrid() { }
+    /**
+     * Updates the positions of the handles and the width of the bar
+     * @param {number} percent
+     * @param {number} index
+     */
+    updateSliderPosition(percent, index) { }
+    // OVERRIDE END

```

Comment:
Same.

Reply:
Added errors for each method if they are not overridden.